src/Symfony/Component/HttpClient/ApiClient.php

```diff
@@ -0,0 +1,136 @@
+use Symfony\Contracts\HttpClient\ApiResponseInterface;
+use Symfony\Contracts\HttpClient\ApiResponseStreamInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/**
+ * @see ApiClientInterface::OPTIONS_DEFAULTS for available options
+ *
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+final class ApiClient implements ApiClientInterface
+{
+    use HttpClientTrait;
+
+    private $client;
+    private $defaultOptions = [
+        'headers' => [
+            'accept' => ['application/json'],
+        ],
+    ] + ApiClientInterface::OPTIONS_DEFAULTS;
+
+    /**
+     * A factory to instantiate the best possible API client for the runtime.
+     *
+     * @param array $defaultOptions     Default requests' options
+     * @param int   $maxHostConnections The maximum number of connections to a single host
+     *
+     * @see HttpClientInterface::OPTIONS_DEFAULTS for available options
+     */
+    public static function create(array $defaultOptions = [], int $maxHostConnections = 6): ApiClientInterface
+    {
+        return new self(HttpClient::create($defaultOptions, $maxHostConnections));
+    }
+
+    public function __construct(HttpClientInterface $client, array $defaultOptions = [])
+    {
+        $this->client = $client;
+
+        if ($defaultOptions) {
+            [, $this->defaultOptions] = self::prepareRequest(null, null, $defaultOptions, $this->defaultOptions);
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function get(string $url, iterable $options = []): ApiResponseInterface
+    {
+        return $this->requestApi('GET', $url, $options);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function head(string $url, iterable $options = []): ApiResponseInterface
+    {
+        return $this->requestApi('HEAD', $url, $options);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function post(string $url, iterable $options = []): ApiResponseInterface
+    {
+        return $this->requestApi('POST', $url, $options);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function put(string $url, iterable $options = []): ApiResponseInterface
+    {
+        return $this->requestApi('PUT', $url, $options);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function patch(string $url, iterable $options = []): ApiResponseInterface
+    {
+        return $this->requestApi('PATCH', $url, $options);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function delete(string $url, iterable $options = []): ApiResponseInterface
+    {
+        return $this->requestApi('DELETE', $url, $options);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function options(string $url, iterable $options = []): ApiResponseInterface
+    {
+        return $this->requestApi('OPTIONS', $url, $options);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function complete($responses, float $timeout = null): ApiResponseStreamInterface
```

Comment:
This should document it accepts `ApiResponse[]` only

Reply:
added